### ISSUE TITLE
pi4j not unpacking

### ISSUE THREAD

**Issue by martwana** [Pi4J/pi4j-v1#25]
When downloading the 0.0.4 deb from downloads and running the dpkg command, it completes with no error, however the /opt/ dir has no pi4j folder with the contents.

I'm using the hard float wheezy and oracle jdk8. I didint have this issue with the soft float build or jdk7u10

If you are going to tell me the issue is fixed in a 'snapshot' built, can you let me know how to install it properly. 

many thanks

martin

**Comment by savageautomate** [comment]
Hi Martin,

I have not seen or heard of that particular issue before.  While you do need 0.0.5-SNAPSHOT to support the Oracle JDK SE8 JDK, the installer package is the same.  You can try using the newer one, but not sure if it will work for you.

https://oss.sonatype.org/content/groups/public/com/pi4j/pi4j-distribution/0.0.5-SNAPSHOT/

See this page for install / uninstall commands:  
http://pi4j.com/install.html

Make sure to run the uninstall command first before trying to install again.

**Comment by savageautomate** [comment]
Hi Martin,

I encountered a similar issue today where I could not install a Pi4J .deb package on one of my older Pis. 
The problem was that a previous version of Pi4J was not uninstalling cleanly due to a removal script failure. 

I was able to use these commands to force the uninstall.

sudo rm /var/lib/dpkg/info/pi4j.prerm
sudo rm /var/lib/dpkg/info/pi4j.postrm
sudo dpkg --remove --force-remove-reinstreq pi4j
sudo rm -r /opt/pi4j

After this, I was able to install a new Pi4J .deb package using the instructions here:
http://pi4j.com/install.html

To really fix this issue, I should add additional logic to the install/uninstall scripts to handle additional edge cases such as the pi4j directory not existing on uninstall attempts.
